airbyte-webapp/src/components/connection/CatalogTree/next/StreamDetailsPanel/StreamFieldsTable/SyncFieldCell/SyncFieldCell.tsx

```diff
@@ -0,0 +1,76 @@
+import { useCallback } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { CheckBox } from "components/ui/CheckBox";
+import { Tooltip } from "components/ui/Tooltip";
+
+import { SyncSchemaField, SyncSchemaFieldObject } from "core/domain/catalog";
+import { SyncMode, DestinationSyncMode } from "core/request/AirbyteClient";
+import { useConnectionFormService } from "hooks/services/ConnectionForm/ConnectionFormService";
+
+interface SyncFieldCellProps {
+  field: SyncSchemaField;
+  checkIsCursor: (path: string[]) => boolean;
+  checkIsChildFieldCursor: (path: string[]) => boolean;
+  checkIsPrimaryKey: (path: string[]) => boolean;
+  checkIsChildFieldPrimaryKey: (path: string[]) => boolean;
+  isFieldSelected: boolean;
+  handleFieldToggle: (fieldPath: string[], isSelected: boolean) => void;
+  syncMode?: SyncMode;
+  destinationSyncMode?: DestinationSyncMode;
+}
+
+export const SyncFieldCell: React.FC<SyncFieldCellProps> = ({
+  checkIsCursor,
+  checkIsChildFieldCursor,
+  checkIsPrimaryKey,
+  checkIsChildFieldPrimaryKey,
+  isFieldSelected,
+  field,
+  handleFieldToggle,
+  syncMode,
+  destinationSyncMode,
+}) => {
+  const { mode } = useConnectionFormService();
+  const isNestedField = SyncSchemaFieldObject.isNestedField(field);
+  const isCursor = checkIsCursor(field.path);
+  const isChildFieldCursor = checkIsChildFieldCursor(field.path);
+  const isPrimaryKey = checkIsPrimaryKey(field.path);
+  const isChildFieldPrimaryKey = checkIsChildFieldPrimaryKey(field.path);
+  const isDisabled =
+    mode === "readonly" ||
```

Comment:
I'm still able to click on the checkboxes. https://127.0.0.1:3001/workspaces/10c578a0-250f-4d2a-9cf9-192107956407/connections/89114a84-9537-4ce9-8e2e-78e2b24431d9/replication

The logic is a bit off here, but we only disable if there's a tooltip.
```
isDisabled = mode === 'readonly'
showTooltip = isDisabled && mode !=== 'readonly'

-> mode === 'readonly' && mode !=== 'readonly' -> false

Reply:
Good catch, addressed in https://github.com/airbytehq/airbyte/pull/21058/commits/22f08b4e09e9f1de5eaf18982ff46ccef044894b and https://github.com/airbytehq/airbyte/pull/21058/commits/b5769c841062ef3288959dbd67a0f627cecfe335